core/utils/pcap_handle.cc

```diff
@@ -0,0 +1,75 @@
+#include "pcap_handle.h"
+
+PcapHandle::PcapHandle(std::string dev) {
+  if (dev.length() == 0) {
+    return;
+  }
+
+  char errbuf[PCAP_ERRBUF_SIZE];
+  handle_ = pcap_open_live(dev.c_str(), PCAP_SNAPLEN, 1, -1, errbuf);
+
+  if (handle_ == nullptr) {
+    return;
+  }
+
+  int ret = pcap_setnonblock(handle_, 1, errbuf);
+  if (ret != 0) {
+    pcap_close(handle_);
+    return;
+  }
+
+  is_initialized_ = true;
+}
+
+PcapHandle::~PcapHandle() {
+  Reset();
+}
+
+PcapHandle::PcapHandle(PcapHandle&& other) {
+  handle_ = other.handle_;
+  is_initialized_ = other.is_initialized_;
+
+  other.handle_ = nullptr;
+  other.is_initialized_ = false;
+}
+
+PcapHandle& PcapHandle::operator=(PcapHandle&& other) {
+  if (this != &other && is_initialized_) {
```

Comment:
I think this if statement should be broken up as follows:

if (this != &other) {
  if (is_initialized_) {
    Reset();
  }
  ...
}
return *this;

Otherwise, if this is equal to other, then down below it will get nulled out, which I don't think you want to happen.